Evm cmd runner panics

#### System information

Erigon version: `./erigon --version` `erigon version 3.00.0-alpha4-39b62b4f`

Commit hash: 39b62b4fda39510251f6ceac89d81f5a94ff8f50


#### Expected behaviour
The built-in evm runner should run the command. This command is trying to run (benchmark) the given code, against the given input.

#### Actual behaviour
Command fails with a panic (related to MdbxTx/AggTx)

#### Steps to reproduce the behaviour
Go to erigon repository folder 
Build erigon evm using `make evm` and run it, as shown in the Backtrace
You will need to checkout the following too
Checkout https://github.com/jwasinger/eip2537-evm-benchmarks
#### Backtrace

````
$ cd eip2537-evm-benchmarks
$ erigon1/build/bin/evm --codefile bytecodes/bench-one.hex --inputfile benchmark_inputs/g1msm28.hex --prestate ./genesis.json --bench run
panic: type *mdbx.MdbxTx need AggTx method

goroutine 1 [running]:
github.com/erigontech/erigon-lib/state.(*SharedDomains).SetTx(0xc0004f47e0, {0x1c325f0?, 0xc001893400?})
	github.com/erigontech/erigon-lib@v0.0.0-00010101000000-000000000000/state/domain_shared.go:598 +0x193
github.com/erigontech/erigon-lib/state.NewSharedDomains({0x1c325f0, 0xc001893400}, {0x1c2b120, 0xc00077a1a0})
	github.com/erigontech/erigon-lib@v0.0.0-00010101000000-000000000000/state/domain_shared.go:120 +0xbd
main.runCmd(0xc00016d7c0)
	github.com/erigontech/erigon/cmd/evm/runner.go:176 +0x8c6
github.com/urfave/cli/v2.(*Command).Run(0x278f320, 0xc00016d7c0, {0xc000f925f0, 0x1, 0x1})
	github.com/urfave/cli/v2@v2.27.4/command.go:276 +0x97d
github.com/urfave/cli/v2.(*Command).Run(0xc000d25340, 0xc000715900, {0xc00013a000, 0x9, 0x9})
	github.com/urfave/cli/v2@v2.27.4/command.go:269 +0xbb7
github.com/urfave/cli/v2.(*App).RunContext(0xc000609000, {0x1c216e0, 0x28f26c0}, {0xc00013a000, 0x9, 0x9})
	github.com/urfave/cli/v2@v2.27.4/app.go:333 +0x5a5
github.com/urfave/cli/v2.(*App).Run(...)
	github.com/urfave/cli/v2@v2.27.4/app.go:307
main.main()
	github.com/erigontech/erigon/cmd/evm/main.go:197 +0x45

````

--- a/cmd/evm/runner.go
+++ b/cmd/evm/runner.go
@@ -169,13 +169,11 @@ func runCmd(ctx *cli.Context) error {
 	} else {
 		genesisConfig = new(types.Genesis)
 	}
-
 	agg, err := state2.NewAggregator(context.Background(), datadir.New(os.TempDir()), config3.HistoryV3AggregationStep, db, log.New())
 	if err != nil {
 		return err
 	}
 	defer agg.Close()
-
 	tdb, err := temporal.New(db, agg)
 	if err != nil {
 		return err

--- a/cmd/evm/runner.go
+++ b/cmd/evm/runner.go
@@ -24,6 +24,8 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/erigontech/erigon-lib/config3"
+	"github.com/erigontech/erigon-lib/kv/temporal"
 	"io"
 	"math/big"
 	"os"
@@ -157,7 +159,7 @@ func runCmd(ctx *cli.Context) error {
 	} else {
 		debugLogger = logger.NewStructLogger(logconfig)
 	}
-	db := memdb.New("")
+	db := memdb.New(os.TempDir())
 	defer db.Close()
 	if ctx.String(GenesisFlag.Name) != "" {
 		gen := readGenesis(ctx.String(GenesisFlag.Name))
@@ -167,7 +169,18 @@ func runCmd(ctx *cli.Context) error {
 	} else {
 		genesisConfig = new(types.Genesis)
 	}
-	tx, err := db.BeginRw(context.Background())
+
+	agg, err := state2.NewAggregator(context.Background(), datadir.New(os.TempDir()), config3.HistoryV3AggregationStep, db, log.New())
+	if err != nil {
+		panic(err)
+	}
+	defer agg.Close()
+
+	tdb, err := temporal.New(db, agg)
+	if err != nil {
+		panic(err)
+	}
+	tx, err := tdb.BeginRw(context.Background())
 	if err != nil {
 		return err
 	}

--- a/cmd/evm/runner.go
+++ b/cmd/evm/runner.go
@@ -172,13 +172,13 @@ func runCmd(ctx *cli.Context) error {
 
 	agg, err := state2.NewAggregator(context.Background(), datadir.New(os.TempDir()), config3.HistoryV3AggregationStep, db, log.New())
 	if err != nil {
-		panic(err)
+		return err
 	}
 	defer agg.Close()
 
 	tdb, err := temporal.New(db, agg)
 	if err != nil {
-		panic(err)
+		return err
 	}
 	tx, err := tdb.BeginRw(context.Background())
 	if err != nil {

